File: sorts/bubble_sort.py
def bubble_sort(input):
    for i in range(len(input)):
        for j in range(0, len(input)-i-1):
            if input[j] > input[j + 1]:
                input[j], input[j + 1] = input[j + 1], input[j]
    return input


print(bubble_sort([3, 4, 1, 2, 3, 10, 9, 7]))

File: sorts/BubbleSort.java
import java.io.*;
import java.util.*;

/*
  Write a solution to the Bubble Sort.
*/
class Solution {
    public static void bubbleSort(int values[]) {
        for (int i = 0; i < values.length; i++) {
            for (int j = 0; j < values.length - i - 1; j++) {
                if (values[j] > values[j + 1]) {
                    int temp = values[j + 1];
                    values[j + 1] = values[j];
                    values[j] = temp;
                }
            }
        }
    }

    public static int[] generateArray(int size) {
        int array[] = new int[size];
        for (int i = 0; i < array.length; i++) {
            array[i] = (int) (Math.random() * 90) + 10;
        }
        return array;
    }

    public static void main(String args[]) {
        int array[] = generateArray(20);
        bubbleSort(array);
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + " ");
        }
    }
}
